### PR TITLE
Align getRandomSeed() implementation to protocol changes from LIP-46

### LINKED ISSUE #8154: Unit Test Review: Random Module
**NOTE:** The review was done on Feb 15, therefore in case the file gets updated after that, plz make sure that you are checking the appropriate version (so that line numbers are valid..).

### [method.spec.ts](https://github.com/LiskHQ/lisk-sdk/blob/development/framework/test/unit/modules/random/method.spec.ts)

- Line 295: I don't know where value 1000 came from. I guess it should be bounded by the maximum size of `validatorReveals` array (I guess implementation needs also to be updated [here](https://github.com/LiskHQ/lisk-sdk/blob/f45f5fa24858b133841953cbe61286531391e7ff/framework/src/modules/random/utils.ts#L76)).
- Tests of lines 322, 344, 366, 388: There was a [PR](https://github.com/LiskHQ/lips/commit/345afc86e79806d48d869a5af5386dd644e91787) fixing a logic error in `getRandomBytes` function which does not seem to have been incorporated neither in the implementation nor in unit tests. For the implementation, [line 87 of random/utils.ts](https://github.com/LiskHQ/lisk-sdk/blob/f45f5fa24858b133841953cbe61286531391e7ff/framework/src/modules/random/utils.ts#L87) should be modified to replace "<=" by "<" (also I think it could be implemented more efficiently given that validatorsReveal array is sorted by height..). If this change is done, then for all aforementioned tests cases, `numberOfSeeds` should be increased by 1 in order to expect the same outcome. 
- Tests of lines 322, 344, 366, 388 assume that function [bitwiseXOR](https://github.com/LiskHQ/lisk-sdk/blob/f45f5fa24858b133841953cbe61286531391e7ff/framework/src/modules/random/utils.ts#L106) is correctly implemented. Although it is simple, it would be good to check that it works as intended.
- line 426: I don't see where function `generateRandomSeeds` is implemented.

### Other Comments

[getSeedRevealValidity](https://github.com/LiskHQ/lisk-sdk/blob/f45f5fa24858b133841953cbe61286531391e7ff/framework/src/modules/random/utils.ts#L41) could be implemented more efficiently by making use of the fact that `validatorsReveal` is ordered according to height (same as in [the implementation of `isSeedValidInput`](https://github.com/LiskHQ/lisk-sdk/blob/f45f5fa24858b133841953cbe61286531391e7ff/framework/src/modules/random/utils.ts#L20)).

### PR DESCRIPTION
**Pull request by bobanm**
### What was the problem?

This PR resolves #8154. It is related only to the second point about changes in PR related to `getRandomBytes()` and `getRandomSeed()`. The remaining points from this original issue have been tracked by the newly created, isolated issues, because they need to be merged into different branches.

### How was it solved?

- updated `getRandomSeed()` function, as per the [LIP change](https://github.com/LiskHQ/lips/commit/345afc86e79806d48d869a5af5386dd644e91787)
- removed redundant import of `lisk-cryptography` package
- updated unit tests for `getRandomSeed()`
- removed redundant import of protocol spec fixture from the test suite
- removed duplicated symlink to protocol spec fixture
- updated protocol spec fixture generator with the new logic
- re-generated protocol spec fixtures using the updated generator

### How was it tested?

Run the updated test suite, and all tests are passing 👌🏻

### CODE REVIEW

File: framework/src/modules/random/utils.ts [hunk 1]
```diff
@@ -77,14 +73,14 @@ export const getRandomSeed = (
 		throw new Error('Number of seeds cannot be greater than 1000.');
```
Comment (gkoumout) [inline framework/src/modules/random/utils.ts:73]: I guess this checked would be removed in a separate PR?

File: framework/test/unit/modules/random/method.spec.ts [hunk 2]
```diff
@@ -341,9 +340,9 @@ describe('RandomModuleMethod', () => {
 			);
 		});
 
-		it('should return XOR random bytes for height=11, numberOfSeeds=3', async () => {
+		it('should return XOR random bytes for height=11, numberOfSeeds=4', async () => {
```
Comment (gkoumout) [inline framework/test/unit/modules/random/method.spec.ts:343]: This test looks identical to the test of line 365. Am I missing something?

File: protocol-specs/generators/pos_random_seed_generation/index.js [hunk 3]
```diff
@@ -131,7 +131,7 @@ const isValidSeedReveal = (seedReveal, previousSeedReveal) =>
 const selectSeedReveal = ({ fromHeight, toHeight, blocksMap, blocksPerRound }) => {
 	const selected = [];
 
-	for (let i = fromHeight; i >= toHeight; i -= 1) {
+	for (let i = fromHeight - 1; i >= toHeight; i -= 1) {
```
Comment (gkoumout) [inline protocol-specs/generators/pos_random_seed_generation/index.js:134]: Could you explain the logic here? I am not sure I understand.

**Review by gkoumout — COMMENTED** [review]
Looks good to me, just left some small comments.

I am not able to judge the changes related to the `lisk-cryptography` package and files of folder `generator_outputs/dpos_random_seed_generation`, so I assume this will be reviewed within SDK.

File: framework/src/modules/random/utils.ts (hunk 1 above)
Comment (bobanm) [reply framework/src/modules/random/utils.ts:73]: Correct, it is handled in #8575. I haven't selected you as the reviewer of that one, because that change has nothing to do with the protocol.

The main reason to handle it in a separate PR is that this PR is a protocol change, and as such should be merged to `release` branch. The removal of 1000 check is an improvement, which we merge to `development` branch.

File: protocol-specs/generators/pos_random_seed_generation/index.js (hunk 3 above)
Comment (bobanm) [reply protocol-specs/generators/pos_random_seed_generation/index.js:134]: The change introduced in this PR make `getRandomSeed()` to include validator seed reveals that are strictly lower than `height + numberOfSeeds`. That's why it is needed to exclude the highest block from the parameters provided to `selectSeedReveal()` function, so the generator will exclude the highest block from the provided range, as per the LIP change.

File: framework/test/unit/modules/random/method.spec.ts (hunk 2 above)
Comment (shuse2) [reply framework/test/unit/modules/random/method.spec.ts:343]: yea it seems to be the same, the intended difference is this test doesn't include the invalid seed reveal and other one should include the invalid one, thus it has different outcome

**Review by shuse2 — APPROVED** [review]

File: framework/test/unit/modules/random/method.spec.ts (hunk 2 above)
Comment (bobanm) [reply framework/test/unit/modules/random/method.spec.ts:343]: Good catch, @gkoumout. After updating the tests which were failing due to the protocol changes, the two test cases you are referring became identical by coincidence 😄 

I have updated one of the test, to validate random bytes when one seed reveal is invalid.

File: protocol-specs/generators/pos_random_seed_generation/index.js (hunk 3 above)
Comment (gkoumout) [reply protocol-specs/generators/pos_random_seed_generation/index.js:134]: ah ok, I just saw the `i-=1`, makes perfect sense.

**Review by gkoumout — APPROVED** [review]
